### PR TITLE
DAOS-12041 test: Add a fault injection test for container create

### PR DESCRIPTION
**Pull request by ashleypittman**
Skip-build-el8-gcc: true
Skip-build-el8-rpm: true
Skip-build-leap15-rpm: true
Skip-build-leap15-icc: true
Skip-unit-test: true
Skip-unit-test-memcheck: true
Skip-func-test-el8: true
Skip-build-ubuntu20-rpm: true

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

**Comment by ashleypittman** [comment]
@liuxuezhao I think this test should pass, it's running two engines on one node and creating a container with rf:1, the failing log is at:

https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-10689/19/artifact/nlt_logs/el8.fault-injection/dnt_fi_cont-create_logs/dnt_reference___41ihj8.log.bz2

I've cherry-picked your PR for the rf issue and that hasn't help the case at all.

**Comment by liuxuezhao** [comment]
> running two engines on one node and creating a container with rf:1

please note that on master branch, the default fault domain level changed to node now, so in your config you only have one domain, then cannot create container with rf:1 which requires at least two domain.
To fix it, either set the fdom level to rank, or use two nodes.

**Comment by ashleypittman** [comment]
> 

Thank you, how do I set the fdom level to rank?

**Comment by liuxuezhao** [comment]
> Thank you, how do I set the fdom level to rank?

NP, "daos cont  create ... --properties=rd_fac:1,rd_lvl:1" (rd_lvl 1 means RANKS, 2 is NODE, default is NODE on master and RANK on 2.2).

**Comment by ashleypittman** [comment]
> NP, "daos cont create ... --properties=rd_fac:1,rd_lvl:1" (rd_lvl 1 means RANK, 2 is NODE, default is NODE on master and RANK on 2.2).

I assume rd_lvl:1 means the same as rf:1 and I don't need both?

**Comment by liuxuezhao** [comment]
> > NP, "daos cont create ... --properties=rd_fac:1,rd_lvl:1" (rd_lvl 1 means RANK, 2 is NODE, default is NODE on master and RANK on 2.2).
> 
> I assume rd_lvl:1 means the same as rf:1 and I don't need both?

right.

**Comment by ashleypittman** [comment]
Thank you, this test works as expected now although it does take a while so we probably don't want to land this as-is.   I'll try again once the current CI run has finished without your commit in this PR to see how that behaves.

**Review by mjmac — DISMISSED** [review]

**Review by daltonbohning — APPROVED** [review]
LGTM but has a conflict

File: utils/node_local_test.py [hunk 1]
```diff
@@ -4368,8 +4370,33 @@ def get_cmd(cont_id):
...
-    rc = test_cmd.launch()
-    return rc
+    return test_cmd.launch()
+
+
+def test_alloc_cont_create(server, conf, wf):
+    """Run container creation under fault injection.
+
+    This test will create a new uuid per iteration, and the test will then try to create a matching
+    container so this is potentially resource intensive.
```
[the remark this answers is not in the document]
Comment (ashleypittman) [reply utils/node_local_test.py:4380]: Running the test causes one to be created.  At some point with these tests we're going to start hitting resource limits with the number of iterations they take but we're not at that point yet.

**Review by mjmac — APPROVED** [review]

**Review by daltonbohning — APPROVED** [review]

**Pull request merged**